fakeldap.py

```diff
@@ -417,12 +417,21 @@ def _search_s(self, base, scope, filterstr, attrlist, attrsonly):
             return [(base, attrs)]
         elif scope == self.SCOPE_ONELEVEL:
             simple_query_regex = r"\(\w+=.+\)$"  # matches things like (some_attribute=value)
-            r = re.compile(simple_query_regex)
-            if r.match(filterstr) is None:  # only this very simple search is supported
+            r_simple = re.compile(simple_query_regex)
+
+            anding_query_regex = r"\(\&(\(\w+=.+\))+\)$"  # matches things like (&(some_attribute=value)(other_attr=othervalue)), but with any number of "anded" conditions
+            r_anded = re.compile(anding_query_regex)
+
+            if r_simple.match(filterstr) is not None:
+                search_attr_name, search_attr_value = filterstr[1:-1].split('=')
+                return self._multiple_attrs_onelevel_search(base, [(search_attr_name, search_attr_value)])
+            elif r_anded.match(filterstr) is not None:
+                bracketList = filterstr[3:-2] # cut of the `(&(` and `))` --> attr1=val1)(attr2=val2
+                attrs_conditions = [condition.split("=") for condition in bracketList.split(")(")]
+                return self._multiple_attrs_onelevel_search(base, attrs_conditions)
```

Comment:
Can we be consistent in  giving `_multiple_attrs_onelevel_search` a list of tuples?  `attrs_conditions = [condition.split("=") for condition in bracketList.split(")(")]` makes the arg a list of lists rather than tuples. The function's logic may not care but it'd be nice to have a consistent type

Reply:
is this sufficiently?

409a02dc79ec5cadf92af1bd971dee12354c5a9d

```diff
diff --git a/fakeldap.py b/fakeldap.py
index f0967a9..03e8437 100644
--- a/fakeldap.py
+++ b/fakeldap.py
@@ -427,7 +427,7 @@ class MockLDAP(object):
                 return self._multiple_attrs_onelevel_search(base, [(search_attr_name, search_attr_value)])
             elif r_anded.match(filterstr) is not None:
                 bracketList = filterstr[3:-2] # cut of the `(&(` and `))` --> attr1=val1)(attr2=val2
-                attrs_conditions = [condition.split("=") for condition in bracketList.split(")(")]
+                attrs_conditions = [tuple(condition.split("=")) for condition in bracketList.split(")(")]
                 return self._multiple_attrs_onelevel_search(base, attrs_conditions)
             else:
                 raise self.PresetReturnRequiredError('search_s("%s", %d, "%s", "%s", %d)' %
```

Reply:
Yeah that should do it